frontend/packages/dev-console/src/components/NamespacedPage.tsx

```diff
@@ -22,11 +27,27 @@ const NamespacedPage: React.FC<NamespacedPageProps> = ({
   disabled,
   onNamespaceChange,
   hideApplications = false,
+  topologyView = false,
+  topologyListView = false,
   variant = NamespacedPageVariants.default,
 }) => (
   <div className="odc-namespaced-page">
     <NamespaceBar disabled={disabled} onNamespaceChange={onNamespaceChange}>
       {!hideApplications && <ApplicationSelector disabled={disabled} />}
+      {topologyView && (
+        <TopologyToggleIcon
+          url={`/topology-list/ns/${getActiveNamespace()}`}
+          icon={<ListIcon size="md" />}
+          tooltipText="List View"
+        />
+      )}
+      {topologyListView && (
+        <TopologyToggleIcon
+          url={`/topology/ns/${getActiveNamespace()}`}
+          icon={<TopologyIcon size="md" />}
+          tooltipText="Topology View"
+        />
+      )}
```

Comment:
Two things:

1. I'd suggest a single variable to control which one is visible - make an enum of the types that can be shown and render based off of that enum
2. I'd carve this out into it's own component - if we expand the number of views to be had here, this _new component_ will be a one-stop shop for dealing with the rendering.

Reply:
To elaborate on no. 1 - The "Boolean trap" is something often found in component libraries. There are many articles out there about it, but this one below describes it well enough (I couldn't find the one I originally was looking for)
https://spicefactory.co/blog/2019/03/26/how-to-avoid-the-boolean-trap-when-designing-react-components/

They go on to explain alternative ways to handle the boolean problem, but I don't think we need that kind of encapsulation for a view/page.